core/src/main/java/jenkins/security/s2m/DefaultFilePathFilter.java

```diff
@@ -59,6 +62,15 @@ protected boolean op(String op, File f) throws SecurityException {
                     LOGGER.log(Level.FINE, "agent allowed to {0} {1}", new Object[] {op, f});
                     return true;
                 } else {
+                    try {
+                        Field current = Request.class.getDeclaredField("CURRENT");
+                        current.setAccessible(true);
+                        Field createdAt = Command.class.getDeclaredField("createdAt");
+                        createdAt.setAccessible(true);
+                        LOGGER.log(Level.WARNING, "Permitting agent-to-controller " + op + " on " + f, (Throwable) createdAt.get(((ThreadLocal) current.get(null)).get()));
+                    } catch (Exception x) {
+                        LOGGER.log(Level.WARNING, null, x);
```

Comment:
We know both `op` and `f` here, so a better error message would probably be useful? Unless we expect this to never happen?

```suggestion
                        LOGGER.log(Level.WARNING, "Permitting agent-to-controller '" + op + "' on '" + f + "', but failed to identify the caller. See https://www.jenkins.io/redirect/permitted-agent-to-controller-file-access", x);
```

Reply:
If you get an error _here_, the `file` and `op` are beside the point—something bigger is broken.